From 0ec4f998cecaeb2ce9ed874c858cbb3b3056cbbe Mon Sep 17 00:00:00 2001
From: Harikrushna Parmar <harikrushna.parmar@nxp.com>
Date: Mon, 2 Sep 2024 10:19:44 +0530
Subject: [PATCH 4/6] MGG-1402 [#ccc] svgpathtools:bugfix: Skip the end path in
 a polyline if no dedicated end path is provided

Introduce an 'is_polygon' flag in 'polygon2pathd' function to handle the end path
in polyline and polygon

Signed-off-by: Harikrushna Parmar <harikrushna.parmar@nxp.com>
---
 svgpathtools/svg_to_paths.py | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/svgpathtools/svg_to_paths.py b/svgpathtools/svg_to_paths.py
index f1a11c2..0d5f47e 100644
--- a/svgpathtools/svg_to_paths.py
+++ b/svgpathtools/svg_to_paths.py
@@ -87,13 +87,13 @@ def polyline2pathd(polyline, is_polygon=False):
     return d
 
 
-def polygon2pathd(polyline):
+def polygon2pathd(polyline, is_polygon):
     """converts the string from a polygon points-attribute to a string 
     for a Path object d-attribute.
     Note:  For a polygon made from n points, the resulting path will be
     composed of n lines (even if some of these lines have length zero).
     """
-    return polyline2pathd(polyline, True)
+    return polyline2pathd(polyline, is_polygon)
 
 
 def rect2pathd(rect):
@@ -214,7 +214,7 @@ def svg2paths(svg_file_location,
     # path strings, add to list
     if convert_polygons_to_paths:
         pgons = [dom2dict(el) for el in doc.getElementsByTagName('polygon')]
-        d_strings += [polygon2pathd(pg) for pg in pgons]
+        d_strings += [polygon2pathd(pg, True) for pg in pgons]
         attribute_dictionary_list += pgons
 
     if convert_lines_to_paths:
-- 
2.34.1

